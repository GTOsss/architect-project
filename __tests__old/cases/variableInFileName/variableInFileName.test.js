const appRoot = process.cwd();
const { resolve } = require('path');
const {
  returnOutputPathsWithContent,
  promisifyCliCommand,
  returnOutputPaths,
  SnapshotSerializer,
} = require('../../utils/utilsForArcStart');
const cleanOutPutBeforeTest = require('../../utils/cleanOutPutBeforeTest');

const { pathForCommand, casesOutputPath } = require('../casesConfigPath');

const currentCommand = pathForCommand.variableInFileName;

describe('variable', () => {
  cleanOutPutBeforeTest(casesOutputPath.variableInFileName);

  test('get path from variableInFileName in template', async () => {
    await promisifyCliCommand(`arc s -c ${currentCommand}`);
    const currentPathToOutput = resolve(appRoot, currentCommand, 'output');

    expect(returnOutputPaths(currentPathToOutput)).toMatchSnapshot();
  });
  test('get content variableInFileName in template', async () => {
    await promisifyCliCommand(`arc s -c ${currentCommand}`);
    const currentPathToOutput = resolve(appRoot, currentCommand, 'output');

    SnapshotSerializer();

    expect(returnOutputPathsWithContent(currentPathToOutput)).toMatchSnapshot();
  });
});
