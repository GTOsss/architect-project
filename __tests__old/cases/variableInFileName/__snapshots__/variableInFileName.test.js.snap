// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`variable get content variableInFileName in template 1`] = `
Object {
  "src": Array [
    Object {
      "content": "const variable = 'myName';",
      "path": "*/variableInFileName/output/src/icons/myName/dirName-myName_test_fox/test-myName-fox.js",
    },
    Object {
      "content": "const variable = 'myName';",
      "path": "*/variableInFileName/output/src/icons/myName/myName-myName-any/myName-myName-any.ts",
    },
    Object {
      "content": "const variable = 'myName';",
      "path": "*/variableInFileName/output/src/icons/myName/myName-myName-any/myName.ts",
    },
  ],
}
`;

exports[`variable get path from variableInFileName in template 1`] = `
Array [
  "{
  folder: 'src',
  files: [
    'C:/Users/mail/Desktop/timoha/arhitect/architect/__tests__/cases/variableInFileName/output/src/icons/myName/dirName-myName_test_fox/test-myName-fox.js',
    'C:/Users/mail/Desktop/timoha/arhitect/architect/__tests__/cases/variableInFileName/output/src/icons/myName/myName-myName-any/myName-myName-any.ts',
    'C:/Users/mail/Desktop/timoha/arhitect/architect/__tests__/cases/variableInFileName/output/src/icons/myName/myName-myName-any/myName.ts'
  ]
}",
]
`;
